+ source /root/mugen-riscv/libs/locallibs/common_lib.sh
++ python3 --version
++ '[' 0 -eq 0 ']'
++ source /root/mugen-riscv/libs/locallibs/common_lib_python.sh
+ main
++ type -t post_test
+ '[' -n function ']'
+ trap post_test EXIT INT HUP TERM
+ rpm -qa
+ grep expect
++ type -t config_params
+ '[' -n '' ']'
++ type -t pre_test
+ '[' -n function ']'
+ pre_test
+ LOG_INFO 'Start to prepare the test environment.'
+ message='Start to prepare the test environment.'
+ python3 /root/mugen-riscv/libs/locallibs/mugen_log.py --level info --message 'Start to prepare the test environment.'
Thu Dec 29 00:30:33 2022 - INFO  - Start to prepare the test environment.
+ DNF_INSTALL 'lxc lxc-devel lxc-libs lxcfs lxcfs-tools tar busybox'
+ pkgs='lxc lxc-devel lxc-libs lxcfs lxcfs-tools tar busybox'
+ node=1
+ '[' -z '' ']'
+ tmpfile=
++ python3 /root/mugen-riscv/libs/locallibs/rpm_manage.py install --pkgs 'lxc lxc-devel lxc-libs lxcfs lxcfs-tools tar busybox' --node 1 --tempfile ''
+ tmpfile2=/tmp/tmpu0nxd3sj
+ '[' -z '' ']'
+ tmpfile=/tmp/tmpu0nxd3sj
++ rpm -qa lxc
++ awk -F - '{print $2}'
+ version=4.0.3
+ LOG_INFO 'End to prepare the test environment.'
+ message='End to prepare the test environment.'
+ python3 /root/mugen-riscv/libs/locallibs/mugen_log.py --level info --message 'End to prepare the test environment.'
Thu Dec 29 00:32:02 2022 - INFO  - End to prepare the test environment.
++ type -t run_test
+ '[' -n function ']'
+ run_test
+ LOG_INFO 'Start to run test.'
+ message='Start to run test.'
+ python3 /root/mugen-riscv/libs/locallibs/mugen_log.py --level info --message 'Start to run test.'
Thu Dec 29 00:32:03 2022 - INFO  - Start to run test.
+ lxc-create -t /usr/share/lxc/templates/lxc-busybox -n myEuler1
chmod: cannot access '/var/lib/lxc/myEuler1/rootfs/bin/passwd': No such file or directory
+ CHECK_RESULT 0 0 0 'Failed to set up container.'
+ actual_result=0
+ expect_result=0
+ mode=0
+ error_log='Failed to set up container.'
+ '[' -z 0 ']'
+ '[' 0 -eq 0 ']'
+ test 0x '!=' 0x
+ return 0
+ lxc-copy --help
+ grep -i Usage
Usage: lxc-copy 
      --usage                      Give a short usage message
+ CHECK_RESULT 0 0 0 'Check lxc-copy --help failed.'
+ actual_result=0
+ expect_result=0
+ mode=0
+ error_log='Check lxc-copy --help failed.'
+ '[' -z 0 ']'
+ '[' 0 -eq 0 ']'
+ test 0x '!=' 0x
+ return 0
+ lxc-copy --usage
+ grep -i 'Usage: lxc-copy'
Usage: lxc-copy [-N|--newname=NEWNAME] [-p|--newpath=NEWPATH] [-R|--rename] [-s|--snapshot] 
+ CHECK_RESULT 0 0 0 'Check lxc-copy --usage failed.'
+ actual_result=0
+ expect_result=0
+ mode=0
+ error_log='Check lxc-copy --usage failed.'
+ '[' -z 0 ']'
+ '[' 0 -eq 0 ']'
+ test 0x '!=' 0x
+ return 0
+ lxc-copy --version
+ grep 4.0.3
4.0.3
+ CHECK_RESULT 0 0 0 'Check lxc-copy --version failed.'
+ actual_result=0
+ expect_result=0
+ mode=0
+ error_log='Check lxc-copy --version failed.'
+ '[' -z 0 ']'
+ '[' 0 -eq 0 ']'
+ test 0x '!=' 0x
+ return 0
+ lxc-copy -s -n myEuler1 -N myEuler2
+ CHECK_RESULT 1 0 0 'Check lxc-copy -s failed.'
+ actual_result=1
+ expect_result=0
+ mode=0
+ error_log='Check lxc-copy -s failed.'
+ '[' -z 1 ']'
+ '[' 0 -eq 0 ']'
+ test 1x '!=' 0x
+ test -n 'Check lxc-copy -s failed.'
+ LOG_ERROR 'Check lxc-copy -s failed.'
+ message='Check lxc-copy -s failed.'
+ python3 /root/mugen-riscv/libs/locallibs/mugen_log.py --level error --message 'Check lxc-copy -s failed.'
Thu Dec 29 00:32:18 2022 - ERROR - Check lxc-copy -s failed.
+ (( exec_result++ ))
+ LOG_ERROR 'oe_test_lxc_device_copy.sh line 39'
+ message='oe_test_lxc_device_copy.sh line 39'
+ python3 /root/mugen-riscv/libs/locallibs/mugen_log.py --level error --message 'oe_test_lxc_device_copy.sh line 39'
Thu Dec 29 00:32:19 2022 - ERROR - oe_test_lxc_device_copy.sh line 39
+ return 0
+ lxc-ls
+ grep myEuler2
+ CHECK_RESULT 1 0 0 'Check lxc-ls failed.'
+ actual_result=1
+ expect_result=0
+ mode=0
+ error_log='Check lxc-ls failed.'
+ '[' -z 1 ']'
+ '[' 0 -eq 0 ']'
+ test 1x '!=' 0x
+ test -n 'Check lxc-ls failed.'
+ LOG_ERROR 'Check lxc-ls failed.'
+ message='Check lxc-ls failed.'
+ python3 /root/mugen-riscv/libs/locallibs/mugen_log.py --level error --message 'Check lxc-ls failed.'
Thu Dec 29 00:32:19 2022 - ERROR - Check lxc-ls failed.
+ (( exec_result++ ))
+ LOG_ERROR 'oe_test_lxc_device_copy.sh line 41'
+ message='oe_test_lxc_device_copy.sh line 41'
+ python3 /root/mugen-riscv/libs/locallibs/mugen_log.py --level error --message 'oe_test_lxc_device_copy.sh line 41'
Thu Dec 29 00:32:20 2022 - ERROR - oe_test_lxc_device_copy.sh line 41
+ return 0
+ lxc-start myEuler1
lxc-start: myEuler1: lxccontainer.c: wait_on_daemonized_start: 961 Received container state "ABORTING" instead of "RUNNING"
myEuler1:tools/lxc_start.c:main:410 starting container process caused "Failed to setup lxc, please check the config file."lxc-start: myEuler1: tools/lxc_start.c: main: 412 The container failed to start
lxc-start: myEuler1: tools/lxc_start.c: main: 415 To get more details, run the container in foreground mode
lxc-start: myEuler1: tools/lxc_start.c: main: 417 Additional information can be obtained by setting the --logfile and --logpriority options
+ CHECK_RESULT 1 0 0 'Failed to start container.'
+ actual_result=1
+ expect_result=0
+ mode=0
+ error_log='Failed to start container.'
+ '[' -z 1 ']'
+ '[' 0 -eq 0 ']'
+ test 1x '!=' 0x
+ test -n 'Failed to start container.'
+ LOG_ERROR 'Failed to start container.'
+ message='Failed to start container.'
+ python3 /root/mugen-riscv/libs/locallibs/mugen_log.py --level error --message 'Failed to start container.'
Thu Dec 29 00:32:21 2022 - ERROR - Failed to start container.
+ (( exec_result++ ))
+ LOG_ERROR 'oe_test_lxc_device_copy.sh line 43'
+ message='oe_test_lxc_device_copy.sh line 43'
+ python3 /root/mugen-riscv/libs/locallibs/mugen_log.py --level error --message 'oe_test_lxc_device_copy.sh line 43'
Thu Dec 29 00:32:22 2022 - ERROR - oe_test_lxc_device_copy.sh line 43
+ return 0
+ lxc-device --help
+ grep -i Usage
Usage: lxc-device --name=NAME -- add|del DEV
      --usage                      Give a short usage message
+ CHECK_RESULT 0 0 0 'Check lxc-device --help failed.'
+ actual_result=0
+ expect_result=0
+ mode=0
+ error_log='Check lxc-device --help failed.'
+ '[' -z 0 ']'
+ '[' 0 -eq 0 ']'
+ test 0x '!=' 0x
+ return 0
+ lxc-device --usage
+ grep -i 'Usage: lxc-device'
Usage: lxc-device [-n|--name=NAME] [-h|--help] [--usage] [--version] 
+ CHECK_RESULT 0 0 0 'Check lxc-device --usage failed.'
+ actual_result=0
+ expect_result=0
+ mode=0
+ error_log='Check lxc-device --usage failed.'
+ '[' -z 0 ']'
+ '[' 0 -eq 0 ']'
+ test 0x '!=' 0x
+ return 0
+ grep 4.0.3
+ lxc-device --version
4.0.3
+ CHECK_RESULT 0 0 0 'Check lxc-device --version failed.'
+ actual_result=0
+ expect_result=0
+ mode=0
+ error_log='Check lxc-device --version failed.'
+ '[' -z 0 ']'
+ '[' 0 -eq 0 ']'
+ test 0x '!=' 0x
+ return 0
+ LOG_INFO 'End to run test.'
+ message='End to run test.'
+ python3 /root/mugen-riscv/libs/locallibs/mugen_log.py --level info --message 'End to run test.'
Thu Dec 29 00:32:22 2022 - INFO  - End to run test.
+ CASE_RESULT 0
+ case_re=0
+ test -z 3
+ test 3 -gt 0
+ LOG_ERROR 'failed to execute the case.'
+ message='failed to execute the case.'
+ python3 /root/mugen-riscv/libs/locallibs/mugen_log.py --level error --message 'failed to execute the case.'
Thu Dec 29 00:32:23 2022 - ERROR - failed to execute the case.
+ exit 3
+ post_test
+ LOG_INFO 'Start to restore the tet environment.'
+ message='Start to restore the tet environment.'
+ python3 /root/mugen-riscv/libs/locallibs/mugen_log.py --level info --message 'Start to restore the tet environment.'
Thu Dec 29 00:32:24 2022 - INFO  - Start to restore the tet environment.
+ lxc-stop myEuler1
lxc-stop: myEuler1: tools/lxc_stop.c: main: 170 myEuler1 is not running
+ lxc-destroy myEuler2
lxc-destroy: myEuler2: tools/lxc_destroy.c: main: 242 Container is not defined
+ lxc-destroy myEuler1
+ DNF_REMOVE
+ node=1
+ pkg_list=
+ mode=0
+ [[ -z /tmp/tmpu0nxd3sj ]]
+ '[' 0 -ne 0 ']'
+ '[' 1 == 0 ']'
+ python3 /root/mugen-riscv/libs/locallibs/rpm_manage.py remove --node 1 --pkgs '' --tempfile /tmp/tmpu0nxd3sj
+ '[' 0 -ne 0 ']'
+ LOG_INFO 'End to restore the tet environment.'
+ message='End to restore the tet environment.'
+ python3 /root/mugen-riscv/libs/locallibs/mugen_log.py --level info --message 'End to restore the tet environment.'
Thu Dec 29 00:32:55 2022 - INFO  - End to restore the tet environment.
